services/api/src/index.ts

```diff
@@ -1,13 +1,44 @@
-import dotenv from 'dotenv';
-dotenv.config();
-
+/* eslint-disable no-console */
 import express from 'express';
 import * as path from 'path';
 import app from './app';
+import WebSocket, { WebSocketServer } from 'ws';
 
 app.use(express.static(path.join(__dirname, '/../../../app/dist')));
 app.get('*', (_req, res) => {
   res.sendFile(path.join(__dirname + '/../../../app/dist/index.html'));
 });
 
-app.listen(process.env.APP_PORT || 8081);
+const server = app.listen(process.env.APP_PORT || 8081);
+
+const wss = new WebSocketServer({
+  noServer: true,
+});
+
+const players = new Map<string, WebSocket>();
+
+server.on('upgrade', (req, socket, head) => {
+  wss.handleUpgrade(req, socket, head, (ws, request) => {
+    wss.emit('connection', ws, request);
+
+    ws.once('message', function message(data, isBinary) {
+      const newUser = data.toString();
```

Comment:
Ar negali cia buti situacijos, kad du skirtingi useriai prisijungs su pvz tuo paciu usernamu ir del to tures ta pati `newUser` objekta? Jeigu gali, tuomet reiketu isivesti kazkoki id, kuris naujam useriui pakurtu uuid ir naudoti ji kaip rakta laikyti connectionui.

Reply:
Kaip minimal alternatyva galim tureti simbolius arba tuscius objektus useriam, kol neturim persistence.
Arba tiesiog rejectint jei vardas jau paimtas.

Reply:
Pagalvojau ir jei liekam su tuo, kad gali but 2 identiski vardai, tai uuid geresnis nei mano proposalai, nes galima ir fronte ir backe ta pati id turet.

Reply:
done